src/Common.cpp

```diff
@@ -0,0 +1,47 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Tooling/CommonOptionsParser.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "../include/BaseTransformation.h"
+#include "../include/Common.h"
+
+using clang::ASTFrontendAction, clang::ASTConsumer, clang::CompilerInstance,
+    clang::StringRef, clang::Rewriter, clang::RecursiveASTVisitor,
+    clang::DeclGroupRef, clang::SourceManager, clang::ASTContext,
+    clang::ASTMutationListener, clang::ASTDeserializationListener;
+using llvm::outs;
+using std::unique_ptr, std::vector, std::move;
+
+// ----------- Frontend Actions ------------ //
+
+BaseFrontendAction::BaseFrontendAction(
+    vector<BaseTransformation> transformations, string output_path):
+        ASTFrontendAction(),
+        transformations(move(transformations)),
+        output_path(move(output_path)) {}
+
+unique_ptr<ASTConsumer> BaseFrontendAction::CreateASTConsumer(
+        CompilerInstance &CI, StringRef file) {
+    rewriter.setSourceMgr(CI.getSourceManager(), CI.getLangOpts());
+    // Seed for the random number engine
+    random_device rd;
+    // Standard mersenne_twister_engine seeded with rd()
+    mt19937 gen(rd());
+    uniform_real_distribution<> dis(0.0, 1.0);
```

Comment:
This function would be called many times, for each file for each augmentation. Fixing the seed and defining a generator should be done once. So, define it in Runner constructor and pass generator as argument in Action constructor.

Reply:
By the way, where do you define seed?